### PR TITLE
Add .gitignore for cabal; Add `unlessE`; Update Quickref.md a bit

### PR DESCRIPTION
**Pull request by lspitzner**

### CODE REVIEW

**Comment by lspitzner** [comment]
the errors have nothing to do with this PR i believe.

**Comment by lspitzner** [comment]
Last change technically incurs major version bump.

File: Quickref.md [hunk 1]
```diff
@@ -48,17 +48,22 @@ Since MonadHold depends on MonadSample, any [S] function also runs in [H] contex
 [ ]   attachPromptlyDynWithMaybe :: (a -> b -> Maybe c) ->  Dynamic a -> Event b -> Event c
 
 -- Combine multiple Events
-[ ]   <>        :: Monoid a => Event a -> Event a -> Event a
+[ ]   <>        ::         Monoid a => Event a -> Event a -> Event a
```
Comment (ryantrinkle) [inline Quickref.md]: This actually has a `Semigroup a` context, not `Monoid a`

https://github.com/reflex-frp/reflex/blob/develop/src/Reflex/Class.hs#L574

File: Quickref.md [hunk 2]
```diff
@@ -48,17 +48,22 @@ Since MonadHold depends on MonadSample, any [S] function also runs in [H] contex
...
+[ ]   unlessE   ::                     Event a -> Event b -> Event a
```
Comment (ryantrinkle) [inline Quickref.md]: Going back to our discussion before, perhaps we should call this `difference` or something in that vein, analogously to [Data.Map.difference](https://hackage.haskell.org/package/containers-0.5.8.1/docs/Data-Map-Strict.html#v:difference)

File: src/Reflex/Class.hs [hunk 3]
```diff
@@ -640,7 +654,7 @@ switchPromptly ea0 eea = do
 
 instance Reflex t => Align (Event t) where
   nil = never
-  align ea eb = fmapMaybe dmapToThese $ merge $ DMap.fromList [LeftTag :=> ea, RightTag :=> eb]
+  align = alignWithMaybe Just
```
Comment (ryantrinkle) [inline src/Reflex/Class.hs]: Awesome!

File: src/Reflex/Class.hs [hunk 4]
```diff
@@ -703,6 +718,19 @@ distributeListOverDyn = distributeListOverDynWith id
 distributeListOverDynWith :: Reflex t => ([a] -> b) -> [Dynamic t a] -> Dynamic t b
 distributeListOverDynWith f = fmap (f . map (\(Const2 _ :=> Identity v) -> v) . DMap.toList) . distributeDMapOverDynPure . DMap.fromList . map (\(k, v) -> Const2 k :=> v) . zip [0 :: Int ..]
 
+-- | Create a new 'Event' that occurs when the first supplied 'Event' occurs
+-- unless the second supplied 'Event' occurs simultaneously.
+unlessE :: Reflex t => Event t a -> Event t b -> Event t a
+unlessE = alignWithMaybe $ \case { This a -> Just a ; _ -> Nothing }
```
Comment (ryantrinkle) [inline src/Reflex/Class.hs]: I know i used bracketed notation when we were discussing it before, but just for consistency with the rest of the code, let's use a new line for each match in the case statement.

**Comment by ryantrinkle** [comment]
Thanks!

**Pull request merged**